sqldb/sqlc/migrations/000003_invoice_events.up.sql

```diff
@@ -1,59 +1,51 @@
--- invoice_event_types stores the different types of events that can be emitted 
--- for invoices.
+-- invoice_event_types stores the different types of invoice events. 
 CREATE TABLE IF NOT EXISTS invoice_event_types(
     id INTEGER PRIMARY KEY,
 
     description TEXT NOT NULL
 );
 
--- invoice_events stores all events related to the node invoices.
+-- invoice_event_types defines the different types of invoice events.
+INSERT INTO invoice_event_types (id, description) 
+VALUES 
+    -- invoice_created is the event type used when an invoice is created.
+    (0, 'invoice_created'), 
+    -- invoice_canceled is the event type used when an invoice is canceled.
+    (1, 'invoice_canceled'), 
+    -- invoice_settled is the event type used when an invoice is settled.
+    (2, 'invoice_settled'),
+    -- setid_created is the event type used when an AMP sub invoice
+    -- corresponding to the set_id is created.
+    (3, 'setid_created'),
+    -- setid_canceled is the event type used when an AMP sub invoice
+    -- corresponding to the set_id is canceled.
+    (4, 'setid_canceled'),
+    -- setid_settled is the event type used when an AMP sub invoice 
+    -- corresponding to the set_id is settled.
+    (5, 'setid_settled');
+
+
+-- invoice_events stores all major events related to the node's invoices and
+-- AMP sub invoices. This table can be used to create a historical view of what
+-- happened to the node's invoices.
 CREATE TABLE IF NOT EXISTS invoice_events (
-    id INTEGER PRIMARY KEY,
+    id BIGINT PRIMARY KEY,
 
-    -- created_at is the creation time of this event.
-    created_at TIMESTAMP NOT NULL,
-
-    -- invoice_id is the reference to the invoice this event was emitted for.
-    invoice_id INTEGER NOT NULL REFERENCES invoices(id),
-
-    -- htlc_id is the reference to the htlc this event was emitted for, may be 
-    -- null.
-    htlc_id BIGINT REFERENCES invoice_htlcs(htlc_id),
-
-    -- set_id is the reference to the set_id this event was emitted for, may be
-    -- null.
-    set_id BLOB NOT NULL REFERENCES amp_invoice_payments(set_id),
+    -- added_at is the timestamp when this event was added.
+    added_at TIMESTAMP NOT NULL,
 
     -- event_type is the type of this event.
-    event_type INTEGER NOT NULL REFERENCES invoice_event_types(id),
+    event_type INTEGER NOT NULL REFERENCES invoice_event_types(id) ON DELETE CASCADE,
```

Comment:
Can we define `event_type` as an `INTEGER` only and remove the table `invoice_event_types`? I think we can provide the semantics using enums and validate them at the code level. Creating a table, however, would be we'd always need a migration if we want to introduce a new event type.

Reply:
So using something like `invoice_event_types` can give us the flexibility of enforcing the supported enums by directly inserting rows, then referencing that: https://github.com/lightningnetwork/lnd/blob/4eecc8540db1fe320fbbf6670daa2a09303f11c6/sqldb/sqlc/migrations/000003_invoice_events.up.sql#L8-L25

>  Creating a table, however, would be we'd always need a migration if we want to introduce a new event type.

The migration would just be inserting a new record. So something pretty minimal, and a fairly low low risk. Here's an  example of this type of migration: https://github.com/lightninglabs/taproot-assets/blob/main/tapdb/sqlc/migrations/000014_multiverse_tree.up.sql#L42-L49

We also have a migration testing framework in `tapd` we can adopt here as well.